homeassistant/components/shelly/__init__.py

```diff
@@ -270,6 +272,9 @@ async def _async_setup_rpc_entry(hass: HomeAssistant, entry: ShellyConfigEntry)
                 async_create_issue_unsupported_firmware(hass, entry)
                 await device.shutdown()
                 raise ConfigEntryNotReady
+            runtime_data.script_events = await get_rpc_scripts_event_types(
```

Comment:
I think we need a test when an exception is raised for `script_getcode` method.

And for this branch tests for `__init__.py` ends with `ValueError: I/O operation on closed file.` Probably a mock is missing somewhere.